gitpod-io/gitpod
# ws-sync + ws-manager-node = ws-daemon

This PR combines two services running as daemonsets (`ws-sync` and `ws-manager-node`) into one new service named **`ws-daemon`**.
We do that because that split made less and less sense, and combining the two services lets us reduce configuration complexity and easier share code/responsibilities.

I have first renamed `ws-sync` to `ws-daemon`, because ws-sync is the more complex component. After making sure that worked, I then integrated `ws-manager-node` into `ws-daemon`, thereby refactoring the new component's configuration surface.

I've also done some minor renaming of packages, e.g. `resourcegov -> resources`, `hostgov -> hosts`. All in all the packages in ws-daemon are much more sensibly named than they were in ws-sync or ws-manager-node. Other than some naming, I've resisted the urge to "just quickly refactor X", just to keep this reviewable.

Other than renaming `components.wsSync` to `components.wsDaemon` there have been no changes in the `values.yaml` that impact self-hosted or production.

## How to test
- start a workspace without `full_workspace_backup`
- start a workspace with `full_workspace_backup`
- run [cpuburn](https://github.com/patrickmn/cpuburn) and observe `/sys/fs/cgroup/cpu/cpu.cfs_quota_us` to ensure the resource controller (formerly resource governor) is still working
![image](https://user-images.githubusercontent.com/3210701/96145686-c20fca80-0f05-11eb-80eb-b4fc92ecc80a.png)
